Sources/BSON/BSONError.swift

```diff
@@ -42,6 +42,22 @@ public enum BSONError {
     }
 }
 
+extension DecodingError {
+    /// Standardize the errors emitted by BSONValue initializers.
+    internal static func _extendedJSONError(
+        codingPath: [CodingKey] = [],
```

Comment:
This array will always be `[]`, so I think we can omit it.

To loop in @kmahar: we're currently using an empty `codingPath` in the errors thrown as part of the `JSON` -> `BSONvalue` conversion since that stage is technically part of "preprocessing" and the user hasn't requested any keys yet. We could alternatively manage our own `codingPath` exactly as we're doing right now with `keyPath`, but I wasn't sure if that was the accurate thing to do.

Reply:
Yeah, just instead of including it in the error message we'd have it as coding path. We originally opted not to do this because the `codingPath` is typically meant to represent the path that the the user took as part of decoding through keyed containers. The `keyPath` here is just where we are in the document when encountering an error during preprocessing.

Reply:
Ok, yeah I agree with not using the coding path for that, leaving it in the error message makes sense.

Reply:
okay sounds good, I removed it